### ISSUE TITLE
Dynamic Lights Integration

### ISSUE THREAD

**Issue by Darkosto** [TheIllusiveC4/Curios#191]
Hello!

Would you be willing to add support for the dynamic lights mod while wearing a torch in a custom curios slot? I've been working on a datapack to allow players a slot for a torch when exploring but unfortunately the dynamic light functionality is not working. I'm aware this may be a feature request normally asked of the mod and not Curios: https://github.com/AtomicStryker/atomicstrykers-minecraft-mods/issues/403#issuecomment-990734248

Thank you very much!
Darkosto

**Comment by TheIllusiveC4** [comment]
I do not want to include this kind of specific mod integration to keep Curios as lean as possible, so formally I'll be closing this issue for the purposes of this repo.

However, the idea of this intrigues me a bit. And it looks easy enough to accomplish. If you're interested, and since the Dynamic Lights author does not want to accomplish this on their end, I'm fully willing to attempt to create a brand new mod that could be used to bridge the functionality so that dynamic lights can be provided in Curio slots.

**Comment by Darkosto** [comment]
Thanks for your patience on my reply! 

I would absolutely love a mod that would create that bridge. What can I do to help out?

**Comment by TheIllusiveC4** [comment]
Nothing much, I just ask for some patience as I'm juggling a lot right now so I don't know when I'll release it. But here's a preview of what I already have just so you know it actually will be done:
![curiouslights](https://user-images.githubusercontent.com/29991504/146488090-1fb93e1b-895a-485e-afdb-448fc344b47c.gif)

Also, what version are looking to use this on? I would likely release it for 1.16 and 1.18, but I'd like to prioritize whichever version you're on since you are the one who asked for this functionality.

**Comment by Darkosto** [comment]
This looks great so far! I'm very excited to try it out. 

1.18.1 would be my preferred version if possible. Thank you very much!

**Comment by TheIllusiveC4** [comment]
@Darkosto Apologies for the long wait, that was much longer than I intended. I don't know if you're still interested, but I've finally released the compatibility mod as Curious Lights:

[https://www.curseforge.com/minecraft/mc-mods/curious-lights](https://www.curseforge.com/minecraft/mc-mods/curious-lights)

It should work across all 1.18.x versions. I also decided to include compatibility with Lucent, which is another dynamic lighting mod that I found while I was working on this, just in case you or anyone else wanted to use that dynamic lighting implementation instead.
